TimeOffTracker.WebApi/Controllers/UsersController.cs

```diff
@@ -0,0 +1,120 @@
+﻿using ApiModels.Models;
+using BusinessLogic.Services;
+using Domain.EF_Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TimeOffTracker.WebApi.Exceptions;
+using TimeOffTracker.WebApi.ViewModels;
+
+namespace TimeOffTracker.WebApi.Controllers
+{
+
+    [ApiController]
+    [Route("auth/[controller]")]
+    public class UsersController : BaseController
+    {
+        private readonly UserManager<User> _userManager;
+        private readonly RoleManager<IdentityRole> _roleManager;
+        private readonly UserService _userService;
+        private readonly ILogger<UsersController> _logger;
+
+        public UsersController( UserManager<User> userManager, RoleManager<IdentityRole> roleManager, UserService userService, ILogger<UsersController> logger)
+        {
+            _userManager = userManager;
+            _roleManager = roleManager;
+            _userService = userService;
+            _logger = logger;
+        }
+
+        [HttpGet]
+        [Authorize]
+        public async Task<IEnumerable<UserApiModel>> GetAll()
+        {
+            return await _userService.GetUsers();
+        }
+
+        [HttpGet("[action]")]
+        [Authorize]
+        public async Task<IActionResult> GetById([FromQuery(Name = "id")] int userId)
+        {
+            UserApiModel userModel = await _userService.GetUser(userId);
+            if (userModel != null)
+                return Ok(userModel);
+            else
+                throw new UserNotFoundException($"Can't find user with Id: {userId}");
+        }
+
+        [HttpGet("[action]")]
+        [Authorize]
+        public async Task<IActionResult> GetByRole([FromQuery(Name = "role")] string roleName)
+        {
+            var users = await _userService.GetUsersByRole(roleName);
+
+            if (users.Count() != 0)
+                return Ok(users);
+            else
+                throw new UserNotFoundException($"Can't find users in role: {roleName}");
+        }
+
+        [HttpPut]
+        [Authorize(Roles = "Admin")]
+        public async Task<ActionResult<UserApiModel>> SetUserRole([FromForm] RoleChangeModel model)
```

Comment:
the endpoint should be not only about role changes - it's about whole user update. So use [FromBody] UserApiModel user in params. Map all fileds to Ef model of the user. And do all this stuff inside the UserService, not in the controller